help/Docfiles/BasicProvers.wlog_then.doc

```diff
@@ -0,0 +1,20 @@
+\DOC wlog_then
+
+\TYPE {wlog_then : term quotation -> term quotation list -> thm_tactic -> tactic}
+
+\SYNOPSIS
+Apply a theorem-tactic using a proposition that can be assumed without loss of
+generality.
+
+\DESCRIBE
+Like {wlog_then}, but the theorem {P |- P} is passed to the user-provided
+theorem-tactic instead of {strip_assume_tac}.
+
```

Comment:
Example. Why is this useful?

Reply:
That is explained for the general case in the Description manual. There is no need to repeat it for every tactical that takes a theorem-tactic as an argument.

Reply:
I agree with @marioxcc here. If more explanation or examples are desired, let's have them in future commits - I don't think they are a blocking requirement for this PR.

Reply:
My question was: why is low level interface exposed. Is there a useful application for this low level interface. Or to word it differently, why is `wlog_then` added to the signature?

Reply:
`Like {wlog_then}` should be `Like {wlog_tac}` I believe.